File: javadocs/memoize_v2/src/main/java/com/memoizeapp/Group.java
package org.memoize.group19;

import java.util.ArrayList;

/**
* This class represents a group created by a user.
* @author group 19
*
*/
public class Group {
	/**The name of the group, input by the user.*/
	private String name;
	/**A description of the group, input by the user.*/
	private String description;
	/**A list of users subscribed to this group.*/
	private ArrayList<User> subscribers;
	/**A list of users with admin rights to this group.*/
	private ArrayList<User> admins;
	/**A list of events for this group*/
	private ArrayList<Event> events;
	
	/**
	 * Method for a group to create a new event within itself.
	 * @param info The info for the creation of the event.
	 */
	public void addEvent(String info) {
		
	}
}

File: Django-REST/memoize/app/models/Group.py


# class Group():

# 	def __init__(self, name, description):
# 		self.name = name
# 		self.description = description
# 		self.subscribers = []
# 		self.admins = []
# 		self.events = []

# 	def add_event(info):
# 		return True

from django.db import models

class MemGroup(models.Model):
	"""Creates a group (triggered by user wanting to create a group).

	Args:
	  	name: the name of the group being created.
	  	description: an optional description of the group.
	  	admins: users that are admins to this group
	  	subscribers: users that are subscribed to this group.   
    """
	name = models.CharField(max_length=255)
	description = models.CharField(max_length=1000)
	admins = models.ManyToManyField('auth.User', related_name='mem_groups')
	subscribers = models.ManyToManyField('auth.User', related_name='sub_groups')